sql/core/src/test/scala/org/apache/spark/sql/UDFSuite.scala

```diff
@@ -581,4 +581,40 @@ class UDFSuite extends QueryTest with SharedSparkSession {
       .toDF("col1", "col2")
     checkAnswer(df.select(myUdf(Column("col1"), Column("col2"))), Row(2020) :: Nil)
   }
+
+  test("case class as element type of Seq/Array") {
```

Comment:
can we also test some special cases:
1. the catalyst schema has more fields than the case class. e.g. `struct<key: int, value: string, col: int>` and `case class TestData(key: Int, value: String)`
2. the fields order doesn't match, e.g. `struct<value: string, key: int>` and `case class TestData(key: Int, value: String)`
3. the catalyst schema has missing fields, e.g. `struct<key: int>` and `case class TestData(key: Int, value: String)`